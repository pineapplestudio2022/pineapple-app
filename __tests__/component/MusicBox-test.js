import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {render} from '@testing-library/react-native';
import MusicBox from '../../src/Components/MusicBoxComponent';

const wrapper = ({children}) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('[MusicBox] Test', () => {
  test('MusicBox snapshot', () => {
    jest.useFakeTimers();
    const rendered = render(<MusicBox />, {wrapper}).toJSON();
    expect(rendered).toMatchSnapshot(); //기존 스냅샷과 일치하는지
    expect(rendered).toBeTruthy(); //컴포넌트가 null, undefinded 같이 falsy 한 값을 가지지 않았는지
  });
});
