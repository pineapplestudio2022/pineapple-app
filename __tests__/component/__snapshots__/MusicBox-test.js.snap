// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MusicBox] Test MusicBox snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {},
        Object {
          "borderRadius": 4,
          "height": 229.45432977461448,
          "overflow": "hidden",
          "width": 278.84615384615387,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "backgroundColor": "#1a1b1c80",
              "bottom": 0,
              "height": 82.2870699881376,
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      >
        <Text
          color="muted.700"
          fontFamily="body"
          fontSize="md"
          fontStyle="normal"
          fontWeight="bold"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#404040",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "700",
              },
              Object {
                "color": "#4be3ac",
                "fontSize": 29.83103459486446,
                "position": "absolute",
                "right": 4,
                "textAlign": "right",
                "top": 8,
                "width": 192.3076923076923,
              },
            ]
          }
        />
        <Text
          color="muted.700"
          fontFamily="body"
          fontSize={29.83103459486446}
          fontStyle="normal"
          fontWeight={400}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#404040",
                "fontSize": 29.83103459486446,
                "fontStyle": "normal",
                "fontWeight": "400",
              },
              Object {
                "color": "#fafafa",
                "fontSize": 29.83103459486446,
                "position": "absolute",
                "right": 4,
                "textAlign": "right",
                "top": 26,
                "width": 153.84615384615384,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
